lib/spack/docs/index.rst

```diff
@@ -49,6 +49,12 @@ or refer to the full manual below.
 
 .. toctree::
    :maxdepth: 2
+   :caption: Module File Support
+
+   module_file_support
```

Comment:
Ok actually maybe I spoke too soon.  I thought we were suggesting that this should be buried under a collapsible menu, but I actually agree that it shouldn't be a top-level *section*.  Basically I just want the module support visible from the front page of the docs, at least in the reference manual section.